lib/elixir/lib/calendar.ex

```diff
@@ -793,7 +793,14 @@ defmodule Calendar do
 
   # Year
   defp format_modifiers("Y" <> rest, width, pad, datetime, format_options, acc) do
-    result = datetime.year |> Integer.to_string() |> pad_leading(width, pad)
+    {sign, year} =
+      if datetime.year < 0 do
+        {"-", datetime.year * -1}
+      else
+        {"", datetime.year}
```

Comment:
```suggestion
        {?-, -datetime.year}
      else
        {[], datetime.year}
```

Reply:
This is a micro optimization but those data structures use less memory (micro because it won't matter here at all :D but I thought I would comment).

Reply:
Add suggestion. 🙇